### PR TITLE
Add message flags

### PR DESCRIPTION
**Pull request by diceroll123**
### Summary

Adds Message Flags to Message.

_If you look at the commits individually, you'll notice that I need to learn how to spell, but that's irrelevant._

### Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/enums.py [hunk 1]
```diff
@@ -172,6 +173,11 @@ class MessageType(Enum):
     premium_guild_tier_2       = 10
     premium_guild_tier_3       = 11
 
+class MessageFlags(Enum):
+    crossposted    = 1
+    is_crosspost   = 2
+    suppress_embeds = 4
```
Comment (BluePhoenixGame) [inline discord/enums.py]: 1,2,4? Is it intended? Shouldn't suppress_embeds's value be 3?

Comment (diceroll123) [reply discord/enums.py]: Intended. See: https://discordapp.com/developers/docs/resources/channel#message-object-message-flags

Much like UserFlags.

**Comment by diceroll123** [comment]
Added the ability to do `Message.crosspost()` (What the client calls "Publish"), for completion's sake.

It updates the flags of the message which were added in this PR, so I've included it.

Unable to appropriately test, as it is not meant for bot accounts, and I will fiddle with it no further.

If necessary I will split it up into _another_ PR, change it, or remove it entirely based on feedback.

**Comment by MattBSG** [comment]
Does the discord API really restrict this to non-bot accounts only?

**Comment by diceroll123** [comment]
Yep.

`discord.errors.Forbidden: 403 FORBIDDEN (error code: 20001): Bots cannot use this endpoint`

File: discord/message.py [hunk 2]
```diff
@@ -936,3 +959,28 @@ def _emoji_reaction(emoji):
...
         return await state.http.ack_message(self.channel.id, self.id)
+
+    async def crosspost(self):
+        """|coro|
+
+        Publishes this message to your announcement channel.
+
+        You need the :attr:`~Permissions.manage_messages` permission to use this.
+
+        The user must not be a bot user.
```
Comment (apple502j) [inline discord/message.py]:
```suggestion
        .. note::
            This can only be used by non-bot accounts.
```

**Comment by diceroll123** [comment]
tbh I copied it from the method above, `ack`. These should be updated appropriately in a new PR I'd say.

**Comment by MattBSG** [comment]
Can confirm after testing with an available announcement channel in a partner server I am in. Crossposting with a bot account is restricted as of the moment it appears.
`discord.errors.Forbidden: 403 FORBIDDEN (error code: 20001): Bots cannot use this endpoint`

However, oddly, logging in with a user account token and setting bot flag to false then attempting a crosspost returns:
`discord.errors.Forbidden: 403 FORBIDDEN (error code: 20002): Only bots can use this endpoint`

Not sure exactly what is going on here

**Comment by Rapptz** [comment]
Superseded by #2433

**Pull request closed without merging**